gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -75,15 +75,26 @@ object TransformHints {
   }
 
   def tag(plan: SparkPlan, hint: TransformHint): Unit = {
-    if (isAlreadyTagged(plan)) {
-      if (isNotTransformable(plan) && hint.isInstanceOf[TRANSFORM_SUPPORTED]) {
-        throw new UnsupportedOperationException(
-          s"Plan was already tagged as non-transformable, " +
-            s"cannot mark it as transformable after that: ${plan.toString()}")
+    val mergedHint = getHintOption(plan)
+      .map {
+        case originalHint @ TRANSFORM_UNSUPPORTED(Some(originalReason)) =>
+          hint match {
+            case TRANSFORM_UNSUPPORTED(Some(newReason)) =>
+              untag(plan)
```

Comment:
Can we remove `untag`? Ditto for the below part. I am assuming re-tagging can overwrite the old value. It looks a mutable map is used in Spark to hold tags. Thanks!

Reply:
removed